### PR TITLE
Add instructions on the custom widget FK relationship magic.

### PR DESCRIPTION
**Pull request by laceysanderson**
# Documentation
Issue #720 

## Description
While working on the relationship widget I cam across undocumented functionality for how to set foreign keys using the chado storage api. This PR documents that functionality as originally described in #720.

## Testing?
Build the docs and look at **Developers Guide > Creating a Custom Field > Creating a Custom Widget**. You should see the following added example,

screen shot 2019-02-04 at 8 32 48 pm

### CODE REVIEW

**Comment by laceysanderson** [comment]
😞 @bradfordcondon any idea why the taxonomy importer is failing on a documentation only PR?

Using PHP 7.1
```
There was 1 error:
1) Tests\tripal_chado\TaxonomyImporterTest::testImportExistingTaxonomyLoader
Error: Call to a member function children() on null
/home/travis/build/tripal/drupal/sites/all/modules/tripal/tripal_chado/includes/TripalImporter/TaxonomyImporter.inc:775
/home/travis/build/tripal/drupal/sites/all/modules/tripal/tripal_chado/includes/TripalImporter/TaxonomyImporter.inc:506
/home/travis/build/tripal/drupal/sites/all/modules/tripal/tripal_chado/includes/TripalImporter/TaxonomyImporter.inc:254
/home/travis/build/tripal/drupal/sites/all/modules/tripal/tests/tripal_chado/loaders/TaxonomyImporterTest.php:36
```

Using PHP 7.0
```
There were 2 errors:
1) Tests\tripal_chado\TaxonomyImporterTest::testImportExistingTaxonomyLoader
Error: Call to a member function children() on null
/home/travis/build/tripal/drupal/sites/all/modules/tripal/tripal_chado/includes/TripalImporter/TaxonomyImporter.inc:775
/home/travis/build/tripal/drupal/sites/all/modules/tripal/tripal_chado/includes/TripalImporter/TaxonomyImporter.inc:506
/home/travis/build/tripal/drupal/sites/all/modules/tripal/tripal_chado/includes/TripalImporter/TaxonomyImporter.inc:254
/home/travis/build/tripal/drupal/sites/all/modules/tripal/tests/tripal_chado/loaders/TaxonomyImporterTest.php:36
2) Tests\tripal_chado\TaxonomyImporterTest::testImportOrganismFromTaxID
Exception: Cannot add organism: 
/home/travis/build/tripal/drupal/sites/all/modules/tripal/tripal_chado/includes/TripalImporter/TaxonomyImporter.inc:704
/home/travis/build/tripal/drupal/sites/all/modules/tripal/tripal_chado/includes/TripalImporter/TaxonomyImporter.inc:246
/home/travis/build/tripal/drupal/sites/all/modules/tripal/tests/tripal_chado/loaders/TaxonomyImporterTest.php:67
ERRORS!
Tests: 1269, Assertions: 6799, Errors: 2.
The command "./vendor/bin/phpunit" exited with 2.
```

**Comment by bradfordcondon** [comment]
dang.  We probably need to just axe that test and/or re-do it to mock the NCBI returned response. Without looking into it too carefuly, I'd bet It's failing because it got nothing back from the NCBI server.

**Review by bradfordcondon — APPROVED** [review]
Looks helpful to me, thanks.

File: docs/dev_guide/custom_field/custom_widget.rst [hunk 1]
```diff
@@ -108,6 +108,41 @@ For our ``obi__organism`` example, the drop-down returns the chado organism_id o
...
+    $field_name = $this->field['field_name'];
+    $field_table = $this->instance['settings']['chado_table'];
+    $linker_field = 'chado-' . $field_table . '__organism_id';
+
+    //...
+    // Validate your data here
+    //...
+
+    // If we have all the keys then set the columns as in the obi__organism ex.
+    if ($subject_id && $object_id && $type_id) {
```
Comment (almasaeed2010) [inline docs/dev_guide/custom_field/custom_widget.rst:134]: It's a little disorienting for me to see undefined variables. Maybe we should mention in the comment where they come from in this example? Or at least that they are intentionally undefined?

**Comment by almasaeed2010** [comment]
Other than my little comment above, I think this looks good!

Thanks!

File: docs/dev_guide/custom_field/custom_widget.rst (hunk 1 above)
Comment (laceysanderson) [reply docs/dev_guide/custom_field/custom_widget.rst:134]: Good point! I'll add in a comment 👍

**Comment by bradfordcondon** [comment]
Looks like two approvals to me.  thanks @laceysanderson , merging.

**Comment by laceysanderson** [comment]
Awesome, Thank you both for the reviews ❤️

**Pull request merged**